httpclient5/src/main/java/org/apache/hc/client5/http/impl/async/HttpAsyncClientEventHandlerFactory.java

```diff
@@ -198,7 +198,7 @@ public void onHeaderInput(final HttpConnection connection, final int streamId, f
                         public void onHeaderOutput(final HttpConnection connection, final int streamId, final List<? extends Header> headers) {
                             if (headerLog.isDebugEnabled()) {
                                 for (int i = 0; i < headers.size(); i++) {
-                                    headerLog.debug(id + " >> " + headers.get(i));
```

Comment:
This should be a foreach loop, index-based access to lists is discouraged: separate ticket.